src/index.js

```diff
@@ -47,7 +54,9 @@ async function run() {
   const root = ReactDOM.createRoot(document.querySelector('#app'));
   root.render(
     <React.StrictMode>
-      <SynthetixProvider {...{ walletAddress, connect, signer }}>
+      <SynthetixProvider
+        {...{ walletAddress, connect, provider, signer, tokens: getTokensFromLocalStorage() }}
```

Comment:
```suggestion
        {...{ walletAddress, connect, provider, signer, token: restoreToken(walletAddress) }}
```

Reply:
app does not need to know about any tokens except the one used for current selected address, it is an unnecessary knowledge